opentelemetry-auto-instrumentation/src/opentelemetry/auto_instrumentation/auto_instrumentation.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import environ, execl
+from os.path import dirname, pathsep
+from shutil import which
+from sys import argv
+
+
+def run() -> None:
+
+    bootstrap_dir = dirname(__file__)
+    python_path = environ.get("PYTHONPATH", None)
+
+    if python_path is None:
+        environ["PYTHONPATH"] = bootstrap_dir
+    else:
+        environ["PYTHONPATH"] = pathsep.join([bootstrap_dir, python_path])
+
+    python3 = which(argv[1])
+    execl(python3, python3, *argv[2:])  # type: ignore
```

Comment:
Have you considered using https://docs.python.org/3/library/runpy.html instead?
execl won't be nice on Windows.
Note that you don't even need PYTHONPATH then because you can do the setup right in this interpreter instance.

Reply:
Thanks, using `runpy` now, this effectively avoids having to manipulate `PYTHONPATH`.